nativelink-docs/deployment-examples.mdx

```diff
@@ -0,0 +1,225 @@
+---
+title: Deployment Examples
+description: 'See the capabilities of remote execution and cache'
+---
+
+# AWS Terraform Deployment
+This directory contains a reference/starting point on creating a full AWS terraform deployment of Native Link's cache and remote execution system.
+
+## Prerequisites - Setup Hosted Zone / Base Domain
+You are required to first setup a Route53 Hosted Zone in AWS. This is because we will generate SSL certificates and need a domain to register them under.
+
+1. Login to AWS and go to [Route53](https://console.aws.amazon.com/route53/v2/hostedzones)
+2. Click `Create hosted zone`
+3. Enter a domain (or subdomain) that you plan on using as the name and ensure it's a `Public hosted zone`
+4. Click into the hosted zone you just created and expand `Hosted zone details` and copy the `Name servers`
+5. In the DNS server that your domain is currently hosted under (it may be another Route53 hosted zone) create a new `NS` record with the same domain/subdomain that you used in Step 3. The value should be the `Name servers` from Step 4
+
+It may take a few mins to propagate
+
+## Terraform Setup
+1. [Install terraform](https://www.terraform.io/downloads)
+2. Open terminal and run `terraform init` in this directory
+3. Run `terraform apply -var base_domain=INSERT_DOMAIN_NAME_YOU_SETUP_IN_PREREQUISITES_HERE`
+
+It will take some time to apply, when it is finished everything should be running. The endpoints are:
+```
+CAS: grpcs://cas.INSERT_DOMAIN_NAME_YOU_SETUP_IN_PREREQUISITES_HERE
+Scheduler: grpcs://scheduler.INSERT_DOMAIN_NAME_YOU_SETUP_IN_PREREQUISITES_HERE
+```
+
+As a reference you should be able to compile this project using bazel with something like:
+```sh
+bazel test //... \
+    --remote_cache=grpcs://cas.INSERT_DOMAIN_NAME_YOU_SETUP_IN_PREREQUISITES_HERE \
+    --remote_executor=grpcs://scheduler.INSERT_DOMAIN_NAME_YOU_SETUP_IN_PREREQUISITES_HERE \
+    --remote_instance_name=main
+```
+
+## Server configuration
+![Native Link AWS Terraform Diagram](https://user-images.githubusercontent.com/1831202/176286845-ff683266-3f23-489c-b58a-3eda49e484be.png)
+
+## Instances
+All instances use the same configuration for the AMI. There are technically two AMI's but only because by default this solution will spawn workers for x86 servers and ARM servers, so two AMIs are required.
+
+### CAS
+The CAS is only used as a public interface to the S3 data. All the services will talk to S3 directly, so they don't need to talk to the CAS instance.
+
+#### More optimal configuration
+You can reduce cost and increase reliability by moving the CAS onto the same machine that invokes the remote execution protocol (like bazel). Then point the configuration to `localhost` and it will translate the S3 calls into the Bazel Remote Execution Protocol API.
```

Comment:
```suggestion
You can reduce cost and increase reliability by moving the CAS onto the same machine that invokes the remote execution protocol (like Bazel). Then point the configuration to `localhost` and it will translate the S3 calls into the Bazel's Remote Build Execution Protocol API.
```